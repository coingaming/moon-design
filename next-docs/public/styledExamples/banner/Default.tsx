import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
