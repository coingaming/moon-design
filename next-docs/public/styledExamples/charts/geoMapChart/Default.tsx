// @ts-nocheck
import React from 'react';
import { Filter, GeoMapChart } from '@heathmont/moon-charts';
import { SingleItemSelect } from '@heathmont/moon-components';

const Example = () => {
  const data = [
    { label: 'Estonia', code: 'EST', value: 120 },
    { label: 'Russia', code: 'RUS', value: 100 },
    { label: 'Belarus', code: 'BLR', value: 80 },
    { label: 'Argentina', code: 'ARG', value: 60 },
    { label: 'USA', code: 'USA', value: 40 },
  ];

  const filter = (
    <Filter title="by GGR">
      {() => (
        <SingleItemSelect
          onChange={() => {}}
          value="ggr"
          options={[
            { label: 'Active players', value: 'actives' },
            { label: 'GGR', value: 'ggr' },
            { label: 'Bets', value: 'bets' },
          ]}
        />
      )}
    </Filter>
  );

  return (
    <GeoMapChart
      title="Top Countries"
      data={data}
      filter={filter}
      hasUpdates={true}
      onUpdate={() => {}}
      onShare={() => {}}
      onExpand={() => {}}
    />
  );
};

export default Example;
