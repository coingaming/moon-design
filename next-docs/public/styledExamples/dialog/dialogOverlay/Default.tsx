import React from 'react';
import { DialogOverlay } from '@heathmont/moon-components';
import { Button } from '@heathmont/moon-core';

const Example = () => {
  const [showDialog, setShowDialog] = React.useState(false);

  return (
    <>
      <Button variant="tertiary" onClick={() => setShowDialog(true)}>
        Open Dialog
      </Button>
      <DialogOverlay
        isOpen={showDialog}
        onDismiss={() => setShowDialog(false)}
      />
    </>
  );
};

export default Example;
