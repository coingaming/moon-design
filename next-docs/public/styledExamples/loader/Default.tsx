import React from 'react';
import { Loader } from '@heathmont/moon-core';

const Example = () => <Loader />;

export default Example;
