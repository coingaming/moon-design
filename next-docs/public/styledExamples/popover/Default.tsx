import React, { useCallback, useEffect, useState } from 'react';
import { Popover } from '@heathmont/moon-popover';
import { Footer, Heading, Text, Button } from '@heathmont/moon-core';

const Example = () => {
  const [open, setOpen] = useState(false);

  const HandlerOpen = useCallback(
    (e: any) => {
      e.preventDefault();
      setOpen(false);
    },
    [open]
  );

  return (
    <Popover
      padding={0}
      placement="bottom"
      isOpen={open}
      onVisibilityChange={(status) => {
        setOpen(status);
        console.log('status: ', status);
      }}
      content={
        <div>
          <div className="p-8 space-y-4">
            <Heading size={20}>Agents</Heading>
            <Text size={14}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris
              nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
              reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur.
            </Text>
          </div>

          <Footer
            primButton={
              <Button fullWidth size="small" onClick={HandlerOpen}>
                OK
              </Button>
            }
            secButton={
              <Button
                fullWidth
                variant="secondary"
                size="small"
                onClick={HandlerOpen}
              >
                Cansel
              </Button>
            }
            tertButton={
              <Button
                fullWidth
                variant="tertiary"
                size="small"
                onClick={HandlerOpen}
              >
                Decline
              </Button>
            }
            isDivider
            size="small"
          />
        </div>
      }
    >
      <Button>Toggle Popover</Button>
    </Popover>
  );
};

export default Example;
