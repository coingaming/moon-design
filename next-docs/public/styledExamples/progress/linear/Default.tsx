import React from 'react';
import { ProgressLinear } from '@heathmont/moon-components';

const Example = () => (
  <>
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
    />
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
      height={5}
    />
  </>
);

export default Example;
