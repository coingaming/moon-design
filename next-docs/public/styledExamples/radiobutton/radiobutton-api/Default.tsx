import React from 'react';
import {
  RadioButtonLabel,
  RadioButtonInput,
  ListItem,
} from '@heathmont/moon-core';

const Example = () => (
  <div className="flex flex-col">
    <h4 className="mb-2">Please select your age:</h4>
    <div className="mb-2">
      <RadioButtonLabel id="age1">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="0 - 30" id="age1" />
          }
        >
          0 - 30
        </ListItem>
      </RadioButtonLabel>
    </div>
    <div className="mb-2">
      <RadioButtonLabel id="age2">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="0 - 30" id="age2" />
          }
        >
          31 - 60
        </ListItem>
      </RadioButtonLabel>
    </div>
    <div className="mb-2">
      <RadioButtonLabel id="age3">
        <ListItem
          backgroundColor="goku.100"
          elementRight={
            <RadioButtonInput name="age" ariaLabel="61 - 100" id="age3" />
          }
        >
          31 - 60
        </ListItem>
      </RadioButtonLabel>
    </div>
  </div>
);

export default Example;
