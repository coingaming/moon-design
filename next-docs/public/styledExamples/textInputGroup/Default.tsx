import React from 'react';
import { TextInputGroup } from '@heathmont/moon-core';

const Example = () => {
  const inputProps1 = {
    type: 'text',
    inputSize: 'xlarge',
    placeholder: 'Country',
  };

  const inputProps2 = {
    type: 'text',
    inputSize: 'xlarge',
    placeholder: 'Phone',
  };

  return (
    <div className="flex flex-col lg:flex-row justify-around items-center w-full gap-2">
      <TextInputGroup
        inputProps={{
          input1: inputProps1,
          input2: inputProps2,
        }}
      />
      <TextInputGroup
        inputProps={{
          input1: inputProps1,
          input2: inputProps2,
        }}
        orientation="vertical"
      />
    </div>
  );
};

export default Example;
