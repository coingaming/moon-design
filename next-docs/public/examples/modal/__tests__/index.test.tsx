/**
 * @jest-environment jsdom
 */

import '../../../../__mocks__/ResizeObserver';
import BigContent from '../BigContent';
import Default from '../Default';
import StyledContent from '../StyledContent';
import WithSelect from '../WithSelect';
import { act, cleanup, fireEvent, render } from '@testing-library/react';

const withRtl = (component: JSX.Element) => <div dir="rtl">{component}</div>;

describe('Modal', () => {
  afterEach(cleanup);
  it('renders Default', async () => {
    const tree = render(<Default />);
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders BigContent', async () => {
    const tree = render(<BigContent />);
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders StyledContent', async () => {
    const tree = render(<StyledContent />);
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders WithSelect', async () => {
    const tree = render(<WithSelect />);
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
});

describe('Modal in RTL', () => {
  afterEach(cleanup);
  it('renders Default', async () => {
    const tree = render(withRtl(<Default />));
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders BigContent', async () => {
    const tree = render(withRtl(<BigContent />));
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders StyledContent', async () => {
    const tree = render(withRtl(<StyledContent />));
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
  it('renders WithSelect', async () => {
    const tree = render(withRtl(<WithSelect />));
    await act(() => fireEvent.click(tree.getByRole('button')));
    expect(tree).toMatchSnapshot();
  });
});
