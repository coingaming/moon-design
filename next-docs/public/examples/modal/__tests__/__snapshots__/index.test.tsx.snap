// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal in Dark Theme renders BigContent 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Dark Theme renders Default 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Dark Theme renders StyledContent 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Dark Theme renders WithSelect 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Light Theme renders BigContent 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Light Theme renders Default 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Light Theme renders StyledContent 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in Light Theme renders WithSelect 1`] = `
<button
  className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  type="button"
>
  Open dialog
  <span
    aria-hidden="true"
    className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
  />
</button>
`;

exports[`Modal in RTL renders BigContent 1`] = `
<div
  dir="rtl"
>
  <button
    className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    Open dialog
    <span
      aria-hidden="true"
      className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
    />
  </button>
</div>
`;

exports[`Modal in RTL renders Default 1`] = `
<div
  dir="rtl"
>
  <button
    className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    Open dialog
    <span
      aria-hidden="true"
      className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
    />
  </button>
</div>
`;

exports[`Modal in RTL renders StyledContent 1`] = `
<div
  dir="rtl"
>
  <button
    className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    Open dialog
    <span
      aria-hidden="true"
      className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
    />
  </button>
</div>
`;

exports[`Modal in RTL renders WithSelect 1`] = `
<div
  dir="rtl"
>
  <button
    className="px-4 h-10 py-2 gap-2 text-moon-14 rounded-moon-i-sm relative z-0 flex justify-center items-center font-medium no-underline overflow-hidden whitespace-nowrap select-none transition duration-200 active:scale-90 btn-primary"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  >
    Open dialog
    <span
      aria-hidden="true"
      className="z-[-1] block absolute inset-0 pointer-events-none transition-[background-color_0.2s_ease-in-out]"
    />
  </button>
</div>
`;
